grammar.ebnf

```diff
@@ -14,10 +14,13 @@ variable             ::= '$' identifier;
 keyword              ::= [^=|#{}\[\]()]+;
 builtin              ::= [A-Z_.?-]+;
 number               ::= [0-9]+ ('.' [0-9]+)?;
-member               ::= '*'? '[' keyword ']' __ pattern NL;
-member-list          ::= NL member+;
+member-base          ::= '[' keyword ']' __ pattern NL;
+default-member       ::= '*' member-base
+nondefault-member    ::= member-base
 
-message              ::= identifier __ '=' __ (pattern | pattern member-list | member-list);
+member-list          ::= NL (__ nondefault-member)* (__ default-member) (__ nondefault-member)*
```

Comment:
We'll need to split `member-list` into `traits-list` and `variants-list`, which are both lists of `members`.  Only the `variant-list` should require one `default-member`.